python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,625 @@
+        f_in.close()
+        f_out.close()
+
+    def extract_datm_at(self, file_in, file_out):
+        """
+        Create a DATM dataset at a point.
+        """
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(file_in, "LONGXY", "LATIXY", "lon", "lat")
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lon=self.plon, lat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lat", "lon"])
+
+        # specify dimension order
+        f_out = f_out.transpose(u"scalar", "time", "lat", "lon")
+
+        # update attributes
+        self.update_metadata(f_out)
+        f_out.attrs["Created_from"] = file_in
+
+        # mode 'w' overwrites file
+        f_out.to_netcdf(path=file_out, mode="w")
+        logger.info("Successfully created file : %s", file_out)
+        f_in.close()
+        f_out.close()
+
+    def write_shell_commands(self, file):
+        """
+        writes out xml commands commands to a file (i.e. shell_commands) for single-point runs
+        """
+        # write_to_file surrounds text with newlines
+        with open(file, 'w') as nl_file:
+            self.write_to_file("# Change below line if you move the subset data directory", nl_file)
+            self.write_to_file("./xmlchange {}={}".format(USRDAT_DIR, self.out_dir), nl_file)
+            self.write_to_file("./xmlchange PTS_LON={}".format(str(self.plon)), nl_file)
+            self.write_to_file("./xmlchange PTS_LAT={}".format(str(self.plat)), nl_file)
+            self.write_to_file("./xmlchange MPILIB=mpi-serial", nl_file)
+
+    def write_datm_streams_lines(self, streamname, datmfiles, file):
+        """
+        writes out lines for the user_nl_datm_streams file for a specific DATM stream
+        for using subset DATM data at a single point
+
+        streamname - stream name (e.g. TPQW)
+        datmfiles - comma-separated list (str) of DATM file names
+        file - file connection to user_nl_datm_streams file
+        """
+        self.write_to_file("{}:datafiles={}".format(streamname, ','.join(datmfiles)), file)
+        self.write_to_file("{}:mapalgo=none".format(streamname), file)
+        self.write_to_file("{}:meshfile=none".format(streamname), file)
+
+    def create_datm_at_point(self, datm_tuple: DatmFiles, datm_syr, datm_eyr, datm_streams_file):
+        """
+        Create all of a DATM dataset at a point.
+        """
+        logger.info("----------------------------------------------------------------------")
+        logger.info("Creating DATM files at %s, %s", self.plon.__str__(), self.plat.__str__())
+
+        # --  create data files
+        infile = []
+        outfile = []
+        solarfiles = []
+        precfiles = []
+        tpqwfiles = []
+        for year in range(datm_syr, datm_eyr + 1):
+            ystr = str(year)
+            for month in range(1, 13):
```

Comment:
[ ] We should make a local constant for number of months in a year.

Reply:
I added two constant here as FIRST_MONTH=1 and LAST_MONTH=12 and used them here instead of the ranges.

Now it should look like the following:
```
for month in range(FIRST_MONTH, LAST_MONTH+1):
```